framework/db/src/main/java/com/cloud/utils/db/GenericDaoBase.java

```diff
@@ -1323,6 +1327,11 @@ protected void addJoins(StringBuilder str, Collection<JoinBuilder<SearchCriteria
     public Pair<List<T>, Integer> searchAndDistinctCount(final SearchCriteria<T> sc, final Filter filter) {
         List<T> objects = search(sc, filter, null, false);
         Integer count = getDistinctCount(sc);
+        // Count cannot be 0 if there is at least a result in the list, see CLOUDSTACK-10320
+        if (count == 0 && !objects.isEmpty()) {
```

Comment:
Is it possible for that `objects` variable there to assume a null value? If not, everything seems to be in order here. Otherwise, it might be better to use CollectionUtils.

Reply:
`search` always return a list, never `null`.